crates/services/p2p/src/p2p_service.rs

```diff
@@ -96,15 +99,19 @@ pub struct FuelP2PService {
     /// Swarm handler for FuelBehaviour
     swarm: Swarm<FuelBehaviour>,
 
-    /// Holds the Sender(s) part of the Oneshot Channel from the NetworkOrchestrator
-    /// Once the ResponseMessage is received from the p2p Network
-    /// It will send it to the NetworkOrchestrator via its unique Sender    
-    outbound_requests_table: HashMap<OutboundRequestId, ResponseChannelItem>,
+    /// Holds active outbound requests and associated oneshot channels.
+    /// When we send a request to the p2p network, we add it here. The sender
+    /// must provide a channel to receive the response.
+    /// Whenever a response (or an error) is received from the p2p network,
+    /// the request is removed from this table, and the channel is used to
+    /// send the result to the caller.
+    outbound_requests_pending: HashMap<OutboundRequestId, TypedResponseChannel>,
```

Comment:
I like the descriptive "table". Curious why you changed the name of these fields.

Reply:
Reverted. I personally think "table" is redundant as the data type is already HashMap which is basically a "table". But then again the name is ok for networking contexts.